src/persistent_merkle_tree/Node.zig

```diff
@@ -697,6 +736,15 @@ pub const Id = enum(u32) {
                 path_lefts[next_d_offset..path_len],
                 path_rights[next_d_offset..path_len],
             );
+
+            // unref prev parents if it's not part of the new tree
+            // can only unref after the rebind
+            for (next_d_offset..path_len) |bit_i| {
+                if (right_move[bit_i] and unfinalized_parents_buf[bit_i] != null) {
+                    pool.unref(unfinalized_parents_buf[bit_i].?);
```

Comment:
We want one of two possible outcomes:
- if there's no error, then both the old and new tree exist (and all temporary/orphaned nodes are freed)
- if there's an error, then only the old tree exists (all newly created nodes are freed)

How does this work in these cases?
- In the non-error case, I suppose the "unfinalized parents" that are being unrefed are these temporary/orphaned nodes
- if an error is hit (eg run out of memory) only after several indices have been rebound?
- if an error is hit immediately? it looks like the root node might get (incorrectly) unrefed?